### PR TITLE
js3-week1 Homework done

### PR DESCRIPTION
**Pull request by dhemir17**

### CODE REVIEW

File: homework/index.js [hunk 1]
```diff
@@ -20,8 +19,8 @@
     const elem = document.createElement(name);
     parent.appendChild(elem);
     Object.entries(options).forEach(([key, value]) => {
-      if (key === 'text') {
-        elem.textContent = value;
+      if (key === 'text' || key === 'Repository' || key === 'Description' || key === 'Forks' || key === 'Updated') {
```
Comment (Meno-101) [inline homework/index.js:22]: this might be, but have tried that with a punch of more key? is it messy after that? or you still can read it, I din't know about you, but for me it's a little bit complicated, I would prefar to keep it as simple as possible, perhaps you can look into: `array.includes`

or perhaps keeping the code as it was is good enough, and you can check your conditions where you call it, for a purpose of course, once you are working in 3,000,000 base code you will find it a bit difficult to manage these things, so be aware, one change may ruined the whole thing



so in conclusion: the function you changed called `createAndAppend` in which it's a generic name of creating any thing and you can append it to a parent of your choice.

I will not ask changes but it will be very kind of you to see them, if not in this homework I will be expecting it in the next one.

File: homework/index.js [hunk 2]
```diff
@@ -30,25 +29,31 @@
...
-    createAndAppend('li', ul, { text: repo.name });
+    createAndAppend('li', ul, { Repository: repo.name, Description: repo.description, Forks: repo.forks, Updated: repo.updated_at});
   }
-
+  
   function main(url) {
+    
+    const div = createAndAppend('div', root); // Create HEADER
+    div.innerHTML = "HYF Repositories";
+    div.className += "div-class";
```
Comment (Meno-101) [inline homework/index.js:39]: this can be slightly different, understanding the base code as `createAndAppend`, is very important.
now of course if you have the previous base as 3,000,000 lines you have to now the functionality you are working with only.

File: homework/index.js (hunk 1 above)
Comment (Tjebbee) [reply homework/index.js:22]: Hahaha, I like it :)

**Comment by Tjebbee** [comment]
Hey @dhemir17 , could you please reply on the detailed feedback of @Meno-101 ?

File: Week1/homework/js_exercises/ex-1-makingFriends.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+/*
+I use this code for calling Axios from js file (without html)
+const scriptAxios = document.createElement('script');
+scriptAxios.src = "https://cdn.jsdelivr.net/npm/axios/dist/axios.min.js";
+document.body.prepend(scriptAxios);
+
+but axios.get method gives error for this. So please add Axios script to 
+html manually it works properly.
```
[the remark this answers is not in the document]
Comment (dhemir17) [reply Week1/homework/js_exercises/ex-1-makingFriends.js:8]: ok I think I waste time on it :( thanks

File: Week1/homework/js_exercises/ex-1-makingFriends.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+/*
```
[the remark this answers is not in the document]
Comment (dhemir17) [reply Week1/homework/js_exercises/ex-1-makingFriends.js:1]: html added :)

File: Week1/homework/js_exercises/ex-1-makingFriends.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+I use this code for calling Axios from js file (without html)
+const scriptAxios = document.createElement('script');
+scriptAxios.src = "https://cdn.jsdelivr.net/npm/axios/dist/axios.min.js";
+document.body.prepend(scriptAxios);
```
[the remark this answers is not in the document]
Comment (dhemir17) [reply Week1/homework/js_exercises/ex-1-makingFriends.js:5]: ok :)thanks

File: Week1/homework/js_exercises/ex-2-programmersHumor.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
+/*
+<script src="https://cdn.jsdelivr.net/npm/axios/dist/axios.min.js"></script>
+<script src="ex-2-programmersHumor.js"></script>
+
+These 2 lines of codes must be added to <head> part
```
[the remark this answers is not in the document]
Comment (dhemir17) [reply Week1/homework/js_exercises/ex-2-programmersHumor.js:5]: fix it I put it in html :)

File: Week1/homework/js_exercises/ex-2-programmersHumor.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    console.log(response);
+    const img2 = document.createElement('img');
+    img2.src = response.data.img;
+    document.body.appendChild(img2);
+  })
+  .catch(function (error) {
+    // handle error
+    console.log(error);
+  })
+  .then(function () {
```
[the remark this answers is not in the document]
Comment (dhemir17) [reply Week1/homework/js_exercises/ex-2-programmersHumor.js:42]: It must be 'finally' :( 
it is removed

File: Week1/homework/js_exercises/ex-3-DogGalery.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
+const ul = document.getElementById('imageUl');
+
+const callXlmBtn =  document.getElementById('xlmButton');
+callXlmBtn.addEventListener('click', function(){
+    const xhr = new XMLHttpRequest;
```
[the remark this answers is not in the document]
Comment (dhemir17) [reply Week1/homework/js_exercises/ex-3-DogGalery.js:5]: Fixed :)

File: Week1/homework/js_exercises/ex-3-DogGalery.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    xhr.open('GET', 'https://dog.ceo/api/breeds/image/random', true);
+    xhr.onload = function (){
+        if( this.status == 200) {
+            const dogs = JSON.parse(this.responseText);
+            console.log(dogs);
+            const li = document.createElement('li');
+            li.innerHTML = `<img src=${dogs.message} width = "200px">`
```
[the remark this answers is not in the document]
Comment (dhemir17) [reply Week1/homework/js_exercises/ex-3-DogGalery.js:12]: Thanks for feedBack :)

File: homework/index.js (hunk 1 above)
Comment (dhemir17) [reply homework/index.js:22]: I totaly change the code :) I agree with you it was silly :)

File: homework/index.js (hunk 2 above)
Comment (dhemir17) [reply homework/index.js:39]: You are right :)
